gulpfile.js

```diff
@@ -0,0 +1,163 @@
+        .demand(['gae_devserver_path'])
+        .argv;
+    })
+    .argv;
+var concat = require('gulp-concat');
+var gulp = require('gulp');
+var gulpStartGae = require('./scripts/gulp-start-gae-devserver');
+var gulpUtil = require('gulp-util');
+var manifest = require('./manifest.json');
+var minifyCss = require('gulp-minify-css');
+var path = require('path');
+
+var gaeDevserverPath = argv.gae_devserver_path;
+var params = {
+  admin_host: '0.0.0.0',
+  admin_port: 8000,
+  host: '0.0.0.0',
+  port: 8181
+};
+if (argv.clear_datastore) {
+  params.clear_datastore = true;
+}
+
+if (argv.enable_sendmail) {
+  params.enable_sendmail = argv.enable_sendmail;
+}
+
+// Check if there are enough commands/actions/tasks to run gulp.
+function checkCommands (yargs, argv, numRequired) {
+  if (argv._.length < numRequired) {
+    // Display help(usage) message.
+    console.log(yargs.help());
+    // Stop gulp and exit.
+    process.exit();
+  }
+}
+// There should atleast be minimum of two defined tasks,
+// one being the default task('.').
+checkCommands(yargs, argv, 2)
+
+var isMinificationNeeded = (argv.minify == 'True');
+var frontendDependencies = manifest.dependencies.frontend;
+var cssFilesPath = [];
+var fontFolderPath = [];
+var cssBackgroundPath = [];
+var generatedCssTargetDir = path.join(
+  'third_party', 'generated',
+  isMinificationNeeded ? 'prod' : 'dev', 'css');
+
+for (var dependencyId in frontendDependencies) {
+  var dependency = frontendDependencies[dependencyId];
+  var dependencyDir = dependency.targetDirPrefix + dependency.version;
+  if (dependency.hasOwnProperty('cssFiles')) {
+    dependency.cssFiles.forEach(function(cssFiles) {
+      cssFilesPath.push(path.join(
+        'third_party', 'static', dependencyDir, cssFiles));
+    });
+  }
+  if (dependency.hasOwnProperty('fontsPath')) {
+    var fontPrefix = '*.{eot,woff2,ttf,woff,eof,svg}';
+    fontFolderPath.push(path.join('third_party', 'static', dependencyDir,
+      dependency.fontsPath, fontPrefix));
+  }
+  if (dependency.hasOwnProperty('cssBackgroundImage')) {
+    dependency.cssBackgroundImage.forEach(function(imagePath) {
+      cssBackgroundPath.push(path.join(
+        'third_party', 'static', dependencyDir, imagePath));
+    });
+  }
+}
+gulp.task('generateCss', function() {
+  gulp.src(cssFilesPath)
+    .pipe(isMinificationNeeded ? minifyCss() : gulpUtil.noop())
+    .pipe(concat('third_party.css'))
+    .pipe(gulp.dest(generatedCssTargetDir));
+});
+
+// This task is used to copy all fonts which are used by
+// Bootstrap and font-Awesome to one folder
+gulp.task('copyFonts', function() {
+  gulp.src(fontFolderPath)
+    .pipe(gulp.dest(path.join(
+      'third_party', 'generated',
+      isMinificationNeeded ? 'prod' : 'dev', 'fonts')));
+});
+
+// This is a task which copies backgroun image used by select2.css
+// TODO(Barnabas) find a way of removing this task.
+// It is a bit of a hacky method and does not seems to scale with a program
```

Comment:
It is a bit of a hack, because it depends on the relative location of the CSS background images of a third-party library with respect to the CSS file that uses them. The currently-affected libraries include ...

(was it Bootstrap and select2? I can't remember.)